code/sorting/src/sleep_sort/sleep_sort.cpp

```diff
@@ -1,29 +1,44 @@
-/* Part of Cosmos by OpenGenus Foundation */
-#include <iostream>
-#include <chrono>
-#include <vector>
-#include <thread>
-
-using namespace std;
-
-int main(int argc, char** argv)
-{
-    vector<thread> threads;
-
-    for (auto i = 1; i < argc; ++i)
-    {
-        threads.emplace_back(
-            [i, &argv]()
-        {
-            int arg = stoi(argv[i]);
-            this_thread::sleep_for(chrono::seconds(arg));
-            cout << argv[i] << endl;
-        }
-            );
-    }
-
-    for (auto &thread: threads)
-        thread.join();
-
-    return 0;
-}
+// C implementation of Sleep Sort 
+#include <stdio.h> 
+#include <windows.h> 
+#include <process.h> 
+
+void routine(void *a) 
+{ 
+	int n = *(int *) a; // typecasting from void to int 
+
+	// Sleeping time is proportional to the number 
+	// More precisely this thread sleep for 'n' milliseconds 
+	Sleep(n); 
+
+	// After the sleep, print the number 
+	printf("%d ", n); 
+} 
+
+
+void sleepSort(int arr[], int n) 
+{ 
+	// An array of threads, one for each of the elements 
+	// in the input array 
+	HANDLE threads[n]; 
+
+	// Create the threads for each of the input array elements 
+	for (int i = 0; i < n; i++) 
+		threads[i] = (HANDLE)_beginthread(&routine, 0, &arr[i]); 
+
+	// Process these threads 
+	WaitForMultipleObjects(n, threads, TRUE, INFINITE); 
```

Comment:
this a windows API, so what if the code was compiled for another platform?

Reply:
My bad, I'll fix these issues.